hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileRequest.java

```diff
@@ -156,6 +277,71 @@ public static long getObjIDFromTxId(long id) {
     return new ImmutablePair<>(baseId, maxAvailableId);
   }
 
+
+  /**
+   * Class to return the results from verifyDirectoryKeysInPath.
+   * Includes the list of missing intermediate directories and
+   * the directory search result code.
+   */
+  public static class OMPathInfoV1 {
+    private OMDirectoryResult directoryResult;
+    private String leafNodeName;
+    private long lastKnownParentId;
+    private long leafNodeObjectId;
+    private List<String> missingParents;
+    private List<OzoneAcl> acls;
+
+    public OMPathInfoV1(String leafNodeName, long lastKnownParentId,
+                        List missingParents, OMDirectoryResult result,
+                        List<OzoneAcl> aclList) {
+      this.leafNodeName = leafNodeName;
+      this.lastKnownParentId = lastKnownParentId;
+      this.missingParents = missingParents;
+      this.directoryResult = result;
+      this.acls = aclList;
```

Comment:
Here we can add exisitingKeyPath also it will help in debugging failure scenario for like mkdir fails with FILE_EXISTS or FILE_EXISTS_IN_GIVEN_PATH or other error

Reply:
'keyName' is already available as local variable and am making use of that. How about add it to 'OMPathInfoV1' later based on requirement ?

Reply:
No my point here is lets say when user tries to create file '/a/b/c/file1" it failed with NOT_A_FILE, but I don't know from log error due to which part of the file exist. (As during some S3 debug, I thought this will be useful)
let me know your thoughts.

Reply:
Thanks for the explanation. Yes, will add it.